diem/diem
# [shuffle] replace json rpc with api in account.rs

## Motivation

We want to replace all the hard coded json rpc invocations with dev api calls in accounts.rs. This pr achieves this. 

## Test Plan

1. Run `cargo run -p shuffle -- new <project_path>` and `cargo run -p shuffle -- node`.
2. Run `cargo run -p shuffle -- console <project_path>`. You should get this output:

![image](https://user-images.githubusercontent.com/55404786/142896087-7aee9b05-3f64-4f7d-b46f-d9bee6ecfab1.png)

3. Run `cargo run -p shuffle -- account`.  You should get an output similar to this:

![image](https://user-images.githubusercontent.com/55404786/142896169-afa63b34-fdd0-4a69-8c47-cbf8c496ea2e.png)

4. Rerun `cargo run -p shuffle -- console <project_path>`. You should get this output. Notice that the account address matches the one from the picture in 3. 

![image](https://user-images.githubusercontent.com/55404786/142896263-e68cc6af-4602-4d5b-8d34-3a1ebb387292.png)

5. To check if we can deploy modules from the account,  run `cargo run -p shuffle -- deploy -p <project_path>`.
6. In your console repl, run `await.devapi.modules()`. You should get an output similar to this:

![image](https://user-images.githubusercontent.com/55404786/142896520-64b5dc0d-c4ec-451e-986a-e2bcee38fdc7.png)



## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.
